docs/concepts.md

```diff
@@ -71,6 +72,44 @@ that are specific to a particular Node (like per-process UNIX file descriptors),
 so handle value 3 for Node A identify a different Channel than handle value 3
 for Node B (unless there happens to be a coincidence of numbering).
 
+The messages that are sent over Channels hold two types of content, data and
+handles. The data component is just a collection of bytes, but the handles
+component allows references to underlying Channels to be copied between Nodes.
+When a Node receives a message that contains these channel references, new
+handle values are created for the receiving Node to use to refer to the channel;
+these values will be _different_ than the handle values used by the sending
+Node. (This is analogous to the
+[transfer of file descriptors between different UNIX processes](https://en.wikipedia.org/wiki/File_descriptor#File_descriptors_as_capabilities).)
+
+### Invocations
+
+The _invocation pattern_ is used for several kinds of interaction between Nodes
+in Oak, and helps to illustrate the use of Channels. In this pattern, Node A has
+an invocation channel to Node B, and it gets Node B to do things for it by
+sending an invocation message down this Channel. The invocation message contains
+no data, but instead contains references to two other ephemeral Channels:
+
+- A reference to the read half of a _request Channel_ that is used to send
+  request messages from A to B.
+- A reference to the write half of a _response Channel_ that is used to send
+  response messages from B to A.
+
+<!-- From: -->
+<!-- https://docs.google.com/drawings/d/11TpZ2a1yNU3RSNgz5L7tm6RuMHW04AwhxN-nOsqWPeo -->
+<img src="images/InvocationPattern.png" width="550">
+
+Node A keeps hold of a handle for the write half of the request channel, and
+sends requests to it. It then reads responses from the read half of the response
+channel until the operation is complete, then closes both ephemeral Channel
+handles.
+
+Node B acts on the invocation message by reading one or more request messages
+from the first handle; once it has performed the requested functionality, it
+sends one or more response messages via the second handle, and closes both of
+its ephemeral Channel handles.
+
+<img src="images/InvocationPattern.gif" width="550">
```

Comment:
This is very useful :) 
Does it show one possible ordering of the events? In particular, does Node A always close its write handle to the request channel before Node B does the `channel_read` on its read handle?

Tiny detail: Node B's handles to both the request channel and the response channel appear at the same time. Is that accurate? Perhaps they could appear in two steps.

Reply:
There are a few possible orderings, but there are also some constraints (e.g. A can't drop all its handles to the request channel until after the invocation is sent, otherwise the request channel will be dropped).

Both handles do appear at the same time (`(write_handle, read_handle) = oak::channel_create()`)

Reply:
That makes sense. So I guess Node B can still read from the _request channel_ even after Node A calls `channel_close` on the write half of the _request channel_, if a message is already sent on the channel. 

I might be misunderstanding something here, but I thought Node B does not create any channels itself. It is receiving two handles from the invocation channel. Is that right?

Reply:
Yep.  Node B's first read will give the pending message, then after that the channel is orphaned (it's empty and there's no-one able to write to it).

So yes, B is just receiving references to the pair of channels that were originally created by A.

Reply:
Thanks for the clarification :)